src/fireEvent.ts

```diff
@@ -14,12 +15,14 @@ const isTextInput = (element?: ReactTestInstance) => {
   // We have to test if the element type is either the TextInput component
   // (which would if it is a composite component) or the string
   // TextInput (which would be true if it is a host component)
-  // All queries but the one by testID return composite component and event
-  // if all queries returned host components, since fireEvent bubbles up
+  // All queries return host components but since fireEvent bubbles up
   // it would trigger the parent prop without the composite component check
   return (
     filterNodeByType(element, TextInput) ||
-    filterNodeByType(element, 'TextInput')
+    filterNodeByType(
+      element,
+      getConfig().hostComponentNames?.textInput || 'TextInput'
```

Comment:
Shouldn't this be using `detectHostComponentsNames`? Otherwise it will fail if, user did not trigger detection using some query.

Reply:
It may cause to run detection when it's not needed but there should be relatively few cases where that would happen so it's not really an issue and it makes the code simpler and less bug prone so I think it's a good idea